### PR TITLE
fix CallbackFn segfault

### LINKED ISSUE #2167: segfault on `main` due to incompatible CallbackFn lifetime and new plugin APIs

### PR DESCRIPTION
**Pull request by Shoeboxam**
Closes #2167

### CODE REVIEW

**Comment by Shoeboxam** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/opendp/opendp/2168?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2168** <a href="https://app.graphite.dev/github/pr/opendp/opendp/2168?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/opendp/opendp/2168?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#2091** <a href="https://app.graphite.dev/github/pr/opendp/opendp/2091?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2 other dependent PRs ([#2095](https://github.com/opendp/opendp/pull/2095) <a href="https://app.graphite.dev/github/pr/opendp/opendp/2095?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#2153](https://github.com/opendp/opendp/pull/2153) <a href="https://app.graphite.dev/github/pr/opendp/opendp/2153?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#2089** <a href="https://app.graphite.dev/github/pr/opendp/opendp/2089?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: python/test/test_core.py [hunk 1]
```diff
@@ -204,13 +204,14 @@ def test_new_domain():
     assert not not_null_domain.member(float("nan"))
 
 
-def test_user_domain():
+@pytest.mark.parametrize("new_domain", [dp.user_domain, _extrinsic_domain])
+def test_user_domain(new_domain):
```
Comment (mccalluc) [inline python/test/test_core.py:208]:
```suggestion
def test_domain(new_domain):
```

File: python/test/test_core.py [hunk 2]
```diff
@@ -279,13 +280,14 @@ def test_extrinsic_free():
     # this test will pass if Queryable extends the lifetime of [] by holding a reference to it
 
 
-def test_user_distance():
+@pytest.mark.parametrize("new_distance,new_divergence", zip([dp.user_distance, _extrinsic_distance], [dp.user_divergence, _extrinsic_divergence]))
+def test_user_distance(new_distance, new_divergence):
```
Comment (mccalluc) [inline python/test/test_core.py:284]:
```suggestion
def test_distance(new_distance, new_divergence):
```

File: python/test/test_core.py [hunk 3]
```diff
@@ -313,6 +315,12 @@ def test_user_distance():
     assert meas.map(2.0)(3.0) == 12.0
 
 
+def test_pure_function():
+    from opendp._internal import _new_pure_function
```
Comment (mccalluc) [inline python/test/test_core.py:319]: Any downside to importing this at the top?

File: python/test/test_core.py (hunk 2 above)
Comment (mccalluc) [reply python/test/test_core.py:284]: `dp.user_distance` is repeated: Unless you intended for it to vary, but forgot, maybe move it inside the function and simplify the signature?

**Review by mccalluc — DISMISSED** [review]
Seems like the parameters on `test_user_distance` could be simplified, but I'll go ahead and mark as approved if you want to get a quick fix in for the build. Also happy to re-review after tweaks.

**Pull request merged**